learn/views.py

```diff
@@ -353,3 +356,73 @@ def autocompleteSuggestionTopic(request):
 			'list': filter_topic,
 		}
 		return JsonResponse(data)
+
+
+url_entires = {}
+
+def youtubeResource(request):
+	if request.method == "POST":
+		if request.POST.get("next") == "Yes":	
+			items = url_entires['session_variable']
+
+			contentDetails = items["contentDetails"]
+			snippet = items['snippet']
+			statistics = items['statistics']
+			m,s = contentDetails["duration"].replace("PT","").replace("S","").split("M")
+			dur = int(s) + int(m)*60 	
+			result = {
+				"published_at"	:	snippet["publishedAt"].replace("T"," ").replace("Z",""),
+				"title"			:	snippet['title'],
+				"description"	:	snippet['description'],
+				"thumbnail_url"	:	snippet['thumbnails']['high']['url'],
+				"duration"		:	dur,
+				"view_count"	:	int(statistics['viewCount']),
+				"like_count"	:	int(statistics['likeCount']),
+				"dislike_count"	:	int(statistics['dislikeCount']),
+				"comment_count"	:	int(statistics['commentCount'])
+			}
+			del url_entires['session_variable']
+			resource = YoutubeResource(**result)
+			resource.save()
+
+		elif request.POST.get("no") == "No":
+			pass
+	return HttpResponseRedirect(reverse('youtubeResourcePreview'))
+
+
+def youtubeResourcePreview(request):
+
+	flag = False
+	if request.method == "POST":
+		url = request.POST.get("url")
+		flag = True
+		if "watch" in url:
+			#Identifies that it is a video
+			temp,slug = url.split("watch?v=")
+			import os
```

Comment:
As per PEP8 convention, import should not be inside any function or class. It should be on the top of the file.